src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -203,32 +171,40 @@ private bool ContainsAllElements(IEnumerable<T> collection)
                     return false;
                 }
             }
+
             return true;
         }
 
-        // Do a in order walk on tree and calls the delegate for each node.
-        // If the action delegate returns false, stop the walk.
-        // Return true if the entire tree has been walked.
-        // Otherwise returns false.
+        /// <summary>
+        /// Does an inorder tree walk and calls the delegate for each node.
+        /// </summary>
+        /// <param name="action">
+        /// The delegate to invoke on each node.
+        /// If the delegate returns <c>false</c>, the walk is stopped.
+        /// </param>
+        /// <returns><c>true</c> if the entire tree has been walked; otherwise, <c>false</c>.</returns>
         internal virtual bool InOrderTreeWalk(TreeWalkPredicate<T> action)
         {
             if (_root == null)
             {
                 return true;
             }
 
-            // The maximum height of a red-black tree is 2*lg(n+1).
+            // The maximum height of a red-black tree is 2 * log2(n+1).
             // See page 264 of "Introduction to algorithms" by Thomas H. Cormen
-            // note: this should be logbase2, but since the stack grows itself, we
-            // don't want the extra cost
-            Stack<Node> stack = new Stack<Node>(2 * (int)(SortedSet<T>.Log2(Count + 1)));
+            // Note: It's not strictly necessary to provide the stack capacity, but we don't
+            // want the stack to unnecessarily allocate arrays as it grows.
+
+            var stack = new Stack<Node>(2 * (int)(Log2(Count + 1)));
             Node current = _root;
+
             while (current != null)
             {
                 stack.Push(current);
                 current = current.Left;
             }
-            while (stack.Count != 0)
+
+            while (stack.Count > 0)
```

Comment:
I prefer using > since `stack.Count` will always be positive. I can change it back though.